# **PROJETO ANÁLISE ESTATÍSTICA DO ENEM 2019**

Este projeto analisa os resultados do ENEM no estado de São Paulo do ano de 2019.

Os dados foram extraídos do site do INEP:

http://inep.gov.br/microdados

In [ ]:
import numpy as np
import pandas as pd

In [ ]:
dados = pd.read_csv('/content/drive/MyDrive/Cursos_Udemy/curso_estatistica_python/microdados_enem_2019_sp.csv',
                    sep=';', encoding='iso-8859-1')
# encoding: codificação de caracteres, normalmente utiliza-se o iso-8859-1, utf-8, latin-1)

## Extração e Limpeza dos dados

In [ ]:
pd.options.display.max_columns = 44

In [ ]:
dados.head()

NU_INSCRICAO  CO_MUNICIPIO_RESIDENCIA NO_MUNICIPIO_RESIDENCIA  \
0     1,90E+11                  3546801            Santa Isabel   
1     1,90E+11                  3538204             Pinhalzinho   
2     1,90E+11                  3550308               São Paulo   
3     1,90E+11                  3548708   São Bernardo do Campo   
4     1,90E+11                  3549409    São Joaquim da Barra   

   CO_UF_RESIDENCIA SG_UF_RESIDENCIA  NU_IDADE TP_SEXO  TP_ESTADO_CIVIL  \
0                35               SP        22       M                1   
1                35               SP        19       M                1   
2                35               SP        17       F                1   
3                35               SP        19       M                1   
4                35               SP        37       M                2   

   TP_COR_RACA  TP_NACIONALIDADE  CO_MUNICIPIO_NASCIMENTO  \
0            3                 1                1506500.0   
1            3                 1                1503804.0   
2            3                 1                1501402.0   
3            3                 1                3548708.0   
4            1                 1                3549409.0   

  NO_MUNICIPIO_NASCIMENTO  CO_UF_NASCIMENTO SG_UF_NASCIMENTO  TP_ST_CONCLUSAO  \
0    Santa Izabel do Pará              15.0               PA                1   
1                 Jacundá              15.0               PA                1   
2                   Belém              15.0               PA                2   
3   São Bernardo do Campo              35.0               SP                2   
4    São Joaquim da Barra              35.0               SP                1   

   TP_ANO_CONCLUIU  TP_ESCOLA  TP_ENSINO  IN_TREINEIRO   CO_ESCOLA  \
0                4          1        1.0             0         NaN   
1                2          1        1.0             0         NaN   
2                0          3        1.0             0  15567761.0   
3                0          2        1.0             0  35904958.0   
4                3          1        1.0             0         NaN   

   CO_MUNICIPIO_ESC       NO_MUNICIPIO_ESC  CO_UF_ESC SG_UF_ESC  \
0               NaN                    NaN        NaN       NaN   
1               NaN                    NaN        NaN       NaN   
2         1501402.0                  Belém       15.0        PA   
3         3548708.0  São Bernardo do Campo       35.0        SP   
4               NaN                    NaN        NaN       NaN   

   TP_DEPENDENCIA_ADM_ESC  TP_LOCALIZACAO_ESC  TP_SIT_FUNC_ESC  \
0                     NaN                 NaN              NaN   
1                     NaN                 NaN              NaN   
2                     4.0                 1.0              1.0   
3                     2.0                 1.0              1.0   
4                     NaN                 NaN              NaN   

   TP_PRESENCA_CN  TP_PRESENCA_CH  TP_PRESENCA_LC  TP_PRESENCA_MT  NU_NOTA_CN  \
0               1               1               1               1      5646.0   
1               0               0               0               0         NaN   
2               1               1               1               1      5762.0   
3               1               1               1               1      4022.0   
4               0               0               0               0         NaN   

   NU_NOTA_CH  NU_NOTA_LC  NU_NOTA_MT  TP_LINGUA  TP_STATUS_REDACAO  \
0      5858.0      5929.0      6002.0          0                1.0   
1         NaN         NaN         NaN          1                NaN   
2      6411.0      6343.0      7318.0          0                1.0   
3      4248.0      5096.0      3932.0          1                1.0   
4         NaN         NaN         NaN          1                NaN   

   NU_NOTA_COMP1  NU_NOTA_COMP2  NU_NOTA_COMP3  NU_NOTA_COMP4  NU_NOTA_COMP5  \
0          160.0          200.0          180.0          200.0          200.0   

In [ ]:
#O número total de inscritos em São Paulo foi de 813772, sendo de 15,97% de todo o Brasil que foi de 5095308 inscritos.
dados.shape

(813772, 44)

In [ ]:
# Análise dos tipos de atributos.
# object: strings
# int64: inteiros
# float64: reais
# complex: complexos
dados.dtypes

NU_INSCRICAO                object
CO_MUNICIPIO_RESIDENCIA      int64
NO_MUNICIPIO_RESIDENCIA     object
CO_UF_RESIDENCIA             int64
SG_UF_RESIDENCIA            object
NU_IDADE                     int64
TP_SEXO                     object
TP_ESTADO_CIVIL              int64
TP_COR_RACA                  int64
TP_NACIONALIDADE             int64
CO_MUNICIPIO_NASCIMENTO    float64
NO_MUNICIPIO_NASCIMENTO     object
CO_UF_NASCIMENTO           float64
SG_UF_NASCIMENTO            object
TP_ST_CONCLUSAO              int64
TP_ANO_CONCLUIU              int64
TP_ESCOLA                    int64
TP_ENSINO                  float64
IN_TREINEIRO                 int64
CO_ESCOLA                  float64
CO_MUNICIPIO_ESC           float64
NO_MUNICIPIO_ESC            object
CO_UF_ESC                  float64
SG_UF_ESC                   object
TP_DEPENDENCIA_ADM_ESC     float64
TP_LOCALIZACAO_ESC         float64
TP_SIT_FUNC_ESC            float64
TP_PRESENCA_CN               int64
TP_PRESENCA_CH      

In [ ]:
dados1 = dados.drop(columns=['CO_MUNICIPIO_RESIDENCIA'])

In [ ]:
dados1.head()

NU_INSCRICAO NO_MUNICIPIO_RESIDENCIA  CO_UF_RESIDENCIA SG_UF_RESIDENCIA  \
0     1,90E+11            Santa Isabel                35               SP   
1     1,90E+11             Pinhalzinho                35               SP   
2     1,90E+11               São Paulo                35               SP   
3     1,90E+11   São Bernardo do Campo                35               SP   
4     1,90E+11    São Joaquim da Barra                35               SP   

   NU_IDADE TP_SEXO  TP_ESTADO_CIVIL  TP_COR_RACA  TP_NACIONALIDADE  \
0        22       M                1            3                 1   
1        19       M                1            3                 1   
2        17       F                1            3                 1   
3        19       M                1            3                 1   
4        37       M                2            1                 1   

   CO_MUNICIPIO_NASCIMENTO NO_MUNICIPIO_NASCIMENTO  CO_UF_NASCIMENTO  \
0                1506500.0    Santa Izabel do Pará              15.0   
1                1503804.0                 Jacundá              15.0   
2                1501402.0                   Belém              15.0   
3                3548708.0   São Bernardo do Campo              35.0   
4                3549409.0    São Joaquim da Barra              35.0   

  SG_UF_NASCIMENTO  TP_ST_CONCLUSAO  TP_ANO_CONCLUIU  TP_ESCOLA  TP_ENSINO  \
0               PA                1                4          1        1.0   
1               PA                1                2          1        1.0   
2               PA                2                0          3        1.0   
3               SP                2                0          2        1.0   
4               SP                1                3          1        1.0   

   IN_TREINEIRO   CO_ESCOLA  CO_MUNICIPIO_ESC       NO_MUNICIPIO_ESC  \
0             0         NaN               NaN                    NaN   
1             0         NaN               NaN                    NaN   
2             0  15567761.0         1501402.0                  Belém   
3             0  35904958.0         3548708.0  São Bernardo do Campo   
4             0         NaN               NaN                    NaN   

   CO_UF_ESC SG_UF_ESC  TP_DEPENDENCIA_ADM_ESC  TP_LOCALIZACAO_ESC  \
0        NaN       NaN                     NaN                 NaN   
1        NaN       NaN                     NaN                 NaN   
2       15.0        PA                     4.0                 1.0   
3       35.0        SP                     2.0                 1.0   
4        NaN       NaN                     NaN                 NaN   

   TP_SIT_FUNC_ESC  TP_PRESENCA_CN  TP_PRESENCA_CH  TP_PRESENCA_LC  \
0              NaN               1               1               1   
1              NaN               0               0               0   
2              1.0               1               1               1   
3              1.0               1               1               1   
4              NaN               0               0               0   

   TP_PRESENCA_MT  NU_NOTA_CN  NU_NOTA_CH  NU_NOTA_LC  NU_NOTA_MT  TP_LINGUA  \
0               1      5646.0      5858.0      5929.0      6002.0          0   
1               0         NaN         NaN         NaN         NaN          1   
2               1      5762.0      6411.0      6343.0      7318.0          0   
3               1      4022.0      4248.0      5096.0      3932.0          1   
4               0         NaN         NaN         NaN         NaN          1   

   TP_STATUS_REDACAO  NU_NOTA_COMP1  NU_NOTA_COMP2  NU_NOTA_COMP3  \
0                1.0          160.0          200.0          180.0   
1                NaN            NaN            NaN            NaN   
2                1.0          160.0          120.0          200.0   
3                1.0           80.0          160.0          120.0   
4                NaN            NaN            NaN            NaN   

   NU_NOTA_COMP4  NU_NOTA_COMP5  NU_NOT

In [ ]:
dados1 = dados1.drop(columns=['CO_UF_RESIDENCIA','SG_UF_RESIDENCIA','CO_MUNICIPIO_NASCIMENTO'], )

In [ ]:
dados1.head(40)

NU_INSCRICAO NO_MUNICIPIO_RESIDENCIA  NU_IDADE TP_SEXO  TP_ESTADO_CIVIL  \
0      1,90E+11            Santa Isabel        22       M                1   
1      1,90E+11             Pinhalzinho        19       M                1   
2      1,90E+11               São Paulo        17       F                1   
3      1,90E+11   São Bernardo do Campo        19       M                1   
4      1,90E+11    São Joaquim da Barra        37       M                2   
5      1,90E+11              Embu-Guaçu        17       F                1   
6      1,90E+11     São José dos Campos        28       F                1   
7      1,90E+11     São José dos Campos        25       F                2   
8      1,90E+11               Guarulhos        23       F                1   
9      1,90E+11               São Paulo        19       F                1   
10     1,90E+11   São José do Rio Preto        18       F                1   
11     1,90E+11                  Sumaré        22       F                1   
12     1,90E+11                  Franca        32       M                1   
13     1,90E+11               São Paulo        19       F                1   
14     1,90E+11                   Bauru        18       M                1   
15     1,90E+11             Santo André        20       F                1   
16     1,90E+11                Campinas        20       F                1   
17     1,90E+11               São Paulo        26       M                1   
18     1,90E+11             Santo André        44       F                2   
19     1,90E+11               São Paulo        42       F                1   
20     1,90E+11                  Santos        15       F                1   
21     1,90E+11                Panorama        16       F                1   
22     1,90E+11              Jaguariúna        19       M                1   
23     1,90E+11               São Paulo        17       F                1   
24     1,90E+11   São Bernardo do Campo        32       F                1   
25     1,90E+11                     Poá        57       M                2   
26     1,90E+11             Carapicuíba        21       M                1   
27     1,90E+11                   Guará        17       F                1   
28     1,90E+11                 Cajamar        17       F                1   
29     1,90E+11               São Paulo        50       F                1   
30     1,90E+11                  Osasco        17       F                1   
31     1,90E+11               São Paulo        19       F                1   
32     1,90E+11               São Paulo        19       M                1   
33     1,90E+11               São Paulo        38       F                2   
34     1,90E+11               São Paulo        50       M                3   
35     1,90E+11               São Paulo        18       F                1   
36     1,90E+11                Campinas        17       F                1   
37     1,90E+11               São Paulo        18       F                1   
38     1,90E+11     São José dos Campos        17       M                1   
39     1,90E+11                  Osasco        38       F                1   

    TP_COR_RACA  TP_NACIONALIDADE NO_MUNICIPIO_NASCIMENTO  CO_UF_NASCIMENTO  \
0             3                 1    Santa Izabel do Pará              15.0   
1             3                 1                 Jacundá              15.0   
2             3                 1                   Belém              15.0   
3             3                 1   São Bernardo do Campo              35.0   
4             1                 1    São Joaquim da Barra              35.0   
5             3                 1             Itacoatiara              13.0   
6             1                 1                  Manaus              13.0   
7             1                 1     São José dos Campos              35.0   
8             1                 1               São Paulo              35.0   
9             

In [ ]:
dados1 = dados1.drop(columns=['NO_MUNICIPIO_NASCIMENTO','CO_UF_NASCIMENTO','SG_UF_NASCIMENTO',
                              'TP_ANO_CONCLUIU','TP_ENSINO','CO_MUNICIPIO_ESC','CO_UF_ESC','SG_UF_ESC'],)

In [ ]:
dados1.head()

NU_INSCRICAO NO_MUNICIPIO_RESIDENCIA  NU_IDADE TP_SEXO  TP_ESTADO_CIVIL  \
0     1,90E+11            Santa Isabel        22       M                1   
1     1,90E+11             Pinhalzinho        19       M                1   
2     1,90E+11               São Paulo        17       F                1   
3     1,90E+11   São Bernardo do Campo        19       M                1   
4     1,90E+11    São Joaquim da Barra        37       M                2   

   TP_COR_RACA  TP_NACIONALIDADE  TP_ST_CONCLUSAO  TP_ESCOLA  IN_TREINEIRO  \
0            3                 1                1          1             0   
1            3                 1                1          1             0   
2            3                 1                2          3             0   
3            3                 1                2          2             0   
4            1                 1                1          1             0   

    CO_ESCOLA       NO_MUNICIPIO_ESC  TP_DEPENDENCIA_ADM_ESC  \
0         NaN                    NaN                     NaN   
1         NaN                    NaN                     NaN   
2  15567761.0                  Belém                     4.0   
3  35904958.0  São Bernardo do Campo                     2.0   
4         NaN                    NaN                     NaN   

   TP_LOCALIZACAO_ESC  TP_SIT_FUNC_ESC  TP_PRESENCA_CN  TP_PRESENCA_CH  \
0                 NaN              NaN               1               1   
1                 NaN              NaN               0               0   
2                 1.0              1.0               1               1   
3                 1.0              1.0               1               1   
4                 NaN              NaN               0               0   

   TP_PRESENCA_LC  TP_PRESENCA_MT  NU_NOTA_CN  NU_NOTA_CH  NU_NOTA_LC  \
0               1               1      5646.0      5858.0      5929.0   
1               0               0         NaN         NaN         NaN   
2               1               1      5762.0      6411.0      6343.0   
3               1               1      4022.0      4248.0      5096.0   
4               0               0         NaN         NaN         NaN   

   NU_NOTA_MT  TP_LINGUA  TP_STATUS_REDACAO  NU_NOTA_COMP1  NU_NOTA_COMP2  \
0      6002.0          0                1.0          160.0          200.0   
1         NaN          1                NaN            NaN            NaN   
2      7318.0          0                1.0          160.0          120.0   
3      3932.0          1                1.0           80.0          160.0   
4         NaN          1                NaN            NaN            NaN   

   NU_NOTA_COMP3  NU_NOTA_COMP4  NU_NOTA_COMP5  NU_NOTA_REDACAO Q025  
0          180.0          200.0          200.0            940.0    B  
1            NaN            NaN            NaN              NaN    A  
2          200.0          200.0          200.0            880.0    A  
3          120.0          100.0          100.0            560.0    B  
4            NaN            NaN            NaN              NaN    B

In [ ]:
dados1 = dados1.drop(columns=['TP_DEPENDENCIA_ADM_ESC','TP_LOCALIZACAO_ESC','TP_SIT_FUNC_ESC'],)

In [ ]:
dados1.head()

NU_INSCRICAO NO_MUNICIPIO_RESIDENCIA  NU_IDADE TP_SEXO  TP_ESTADO_CIVIL  \
0     1,90E+11            Santa Isabel        22       M                1   
1     1,90E+11             Pinhalzinho        19       M                1   
2     1,90E+11               São Paulo        17       F                1   
3     1,90E+11   São Bernardo do Campo        19       M                1   
4     1,90E+11    São Joaquim da Barra        37       M                2   

   TP_COR_RACA  TP_NACIONALIDADE  TP_ST_CONCLUSAO  TP_ESCOLA  IN_TREINEIRO  \
0            3                 1                1          1             0   
1            3                 1                1          1             0   
2            3                 1                2          3             0   
3            3                 1                2          2             0   
4            1                 1                1          1             0   

    CO_ESCOLA       NO_MUNICIPIO_ESC  TP_PRESENCA_CN  TP_PRESENCA_CH  \
0         NaN                    NaN               1               1   
1         NaN                    NaN               0               0   
2  15567761.0                  Belém               1               1   
3  35904958.0  São Bernardo do Campo               1               1   
4         NaN                    NaN               0               0   

   TP_PRESENCA_LC  TP_PRESENCA_MT  NU_NOTA_CN  NU_NOTA_CH  NU_NOTA_LC  \
0               1               1      5646.0      5858.0      5929.0   
1               0               0         NaN         NaN         NaN   
2               1               1      5762.0      6411.0      6343.0   
3               1               1      4022.0      4248.0      5096.0   
4               0               0         NaN         NaN         NaN   

   NU_NOTA_MT  TP_LINGUA  TP_STATUS_REDACAO  NU_NOTA_COMP1  NU_NOTA_COMP2  \
0      6002.0          0                1.0          160.0          200.0   
1         NaN          1                NaN            NaN            NaN   
2      7318.0          0                1.0          160.0          120.0   
3      3932.0          1                1.0           80.0          160.0   
4         NaN          1                NaN            NaN            NaN   

   NU_NOTA_COMP3  NU_NOTA_COMP4  NU_NOTA_COMP5  NU_NOTA_REDACAO Q025  
0          180.0          200.0          200.0            940.0    B  
1            NaN            NaN            NaN              NaN    A  
2          200.0          200.0          200.0            880.0    A  
3          120.0          100.0          100.0            560.0    B  
4            NaN            NaN            NaN              NaN    B

In [ ]:
dados1.shape

(813772, 29)

Corrigindo erros nas notas

In [ ]:
dados1.loc[:,'NU_NOTA_CN'] /= 10

In [ ]:
dados1.head(12)

NU_INSCRICAO NO_MUNICIPIO_RESIDENCIA  NU_IDADE TP_SEXO  TP_ESTADO_CIVIL  \
0      1,90E+11            Santa Isabel        22       M                1   
1      1,90E+11             Pinhalzinho        19       M                1   
2      1,90E+11               São Paulo        17       F                1   
3      1,90E+11   São Bernardo do Campo        19       M                1   
4      1,90E+11    São Joaquim da Barra        37       M                2   
5      1,90E+11              Embu-Guaçu        17       F                1   
6      1,90E+11     São José dos Campos        28       F                1   
7      1,90E+11     São José dos Campos        25       F                2   
8      1,90E+11               Guarulhos        23       F                1   
9      1,90E+11               São Paulo        19       F                1   
10     1,90E+11   São José do Rio Preto        18       F                1   
11     1,90E+11                  Sumaré        22       F                1   

    TP_COR_RACA  TP_NACIONALIDADE  TP_ST_CONCLUSAO  TP_ESCOLA  IN_TREINEIRO  \
0             3                 1                1          1             0   
1             3                 1                1          1             0   
2             3                 1                2          3             0   
3             3                 1                2          2             0   
4             1                 1                1          1             0   
5             3                 1                2          2             0   
6             1                 1                1          1             0   
7             1                 1                1          1             0   
8             1                 1                1          1             0   
9             3                 1                1          1             0   
10            1                 1                2          2             0   
11            4                 1                1          1             0   

     CO_ESCOLA       NO_MUNICIPIO_ESC  TP_PRESENCA_CN  TP_PRESENCA_CH  \
0          NaN                    NaN               1               1   
1          NaN                    NaN               0               0   
2   15567761.0                  Belém               1               1   
3   35904958.0  São Bernardo do Campo               1               1   
4          NaN                    NaN               0               0   
5          NaN                    NaN               1               1   
6          NaN                    NaN               0               0   
7          NaN                    NaN               0               0   
8          NaN                    NaN               1               1   
9          NaN                    NaN               0               0   
10  35028927.0  São José do Rio Preto               1               1   
11         NaN                    NaN               0               0   

    TP_PRESENCA_LC  TP_PRESENCA_MT  NU_NOTA_CN  NU_NOTA_CH  NU_NOTA_LC  \
0                1               1       564.6      5858.0      5929.0   
1                0               0         NaN         NaN         NaN   
2                1               1       576.2      6411.0      6343.0   
3                1               1       402.2      4248.0      5096.0   
4                0               0         NaN         NaN         NaN   
5                1               1       399.6      4268.0      4872.0   
6                0               0         NaN         NaN         NaN   
7                0               0         NaN         NaN         NaN   
8                1               1       625.7      6465.0      6293.0   
9                0               0         NaN         NaN         NaN   
10               1               1       449.5      5333.0      5559.0   
11               0               0         NaN         NaN         NaN   

    NU_NOTA_MT  TP_LINGUA  TP_STATUS_REDACAO  N

In [ ]:
dados1.loc[:,'NU_NOTA_CH'] /= 10
dados1.loc[:,'NU_NOTA_LC'] /= 10
dados1.loc[:,'NU_NOTA_MT'] /= 10

In [ ]:
dados1.head()

NU_INSCRICAO NO_MUNICIPIO_RESIDENCIA  NU_IDADE TP_SEXO  TP_ESTADO_CIVIL  \
0     1,90E+11            Santa Isabel        22       M                1   
1     1,90E+11             Pinhalzinho        19       M                1   
2     1,90E+11               São Paulo        17       F                1   
3     1,90E+11   São Bernardo do Campo        19       M                1   
4     1,90E+11    São Joaquim da Barra        37       M                2   

   TP_COR_RACA  TP_NACIONALIDADE  TP_ST_CONCLUSAO  TP_ESCOLA  IN_TREINEIRO  \
0            3                 1                1          1             0   
1            3                 1                1          1             0   
2            3                 1                2          3             0   
3            3                 1                2          2             0   
4            1                 1                1          1             0   

    CO_ESCOLA       NO_MUNICIPIO_ESC  TP_PRESENCA_CN  TP_PRESENCA_CH  \
0         NaN                    NaN               1               1   
1         NaN                    NaN               0               0   
2  15567761.0                  Belém               1               1   
3  35904958.0  São Bernardo do Campo               1               1   
4         NaN                    NaN               0               0   

   TP_PRESENCA_LC  TP_PRESENCA_MT  NU_NOTA_CN  NU_NOTA_CH  NU_NOTA_LC  \
0               1               1       564.6       585.8       592.9   
1               0               0         NaN         NaN         NaN   
2               1               1       576.2       641.1       634.3   
3               1               1       402.2       424.8       509.6   
4               0               0         NaN         NaN         NaN   

   NU_NOTA_MT  TP_LINGUA  TP_STATUS_REDACAO  NU_NOTA_COMP1  NU_NOTA_COMP2  \
0       600.2          0                1.0          160.0          200.0   
1         NaN          1                NaN            NaN            NaN   
2       731.8          0                1.0          160.0          120.0   
3       393.2          1                1.0           80.0          160.0   
4         NaN          1                NaN            NaN            NaN   

   NU_NOTA_COMP3  NU_NOTA_COMP4  NU_NOTA_COMP5  NU_NOTA_REDACAO Q025  
0          180.0          200.0          200.0            940.0    B  
1            NaN            NaN            NaN              NaN    A  
2          200.0          200.0          200.0            880.0    A  
3          120.0          100.0          100.0            560.0    B  
4            NaN            NaN            NaN              NaN    B

Renomeando Variáveis (colunas)

In [ ]:
dados1 = dados1.rename(columns={'NU_NOTA_REDACAO': 'NOTA_REDACAO'})

In [ ]:
dados1.head()

NU_INSCRICAO NO_MUNICIPIO_RESIDENCIA  NU_IDADE TP_SEXO  TP_ESTADO_CIVIL  \
0     1,90E+11            Santa Isabel        22       M                1   
1     1,90E+11             Pinhalzinho        19       M                1   
2     1,90E+11               São Paulo        17       F                1   
3     1,90E+11   São Bernardo do Campo        19       M                1   
4     1,90E+11    São Joaquim da Barra        37       M                2   

   TP_COR_RACA  TP_NACIONALIDADE  TP_ST_CONCLUSAO  TP_ESCOLA  IN_TREINEIRO  \
0            3                 1                1          1             0   
1            3                 1                1          1             0   
2            3                 1                2          3             0   
3            3                 1                2          2             0   
4            1                 1                1          1             0   

    CO_ESCOLA       NO_MUNICIPIO_ESC  TP_PRESENCA_CN  TP_PRESENCA_CH  \
0         NaN                    NaN               1               1   
1         NaN                    NaN               0               0   
2  15567761.0                  Belém               1               1   
3  35904958.0  São Bernardo do Campo               1               1   
4         NaN                    NaN               0               0   

   TP_PRESENCA_LC  TP_PRESENCA_MT  NU_NOTA_CN  NU_NOTA_CH  NU_NOTA_LC  \
0               1               1       564.6       585.8       592.9   
1               0               0         NaN         NaN         NaN   
2               1               1       576.2       641.1       634.3   
3               1               1       402.2       424.8       509.6   
4               0               0         NaN         NaN         NaN   

   NU_NOTA_MT  TP_LINGUA  TP_STATUS_REDACAO  NU_NOTA_COMP1  NU_NOTA_COMP2  \
0       600.2          0                1.0          160.0          200.0   
1         NaN          1                NaN            NaN            NaN   
2       731.8          0                1.0          160.0          120.0   
3       393.2          1                1.0           80.0          160.0   
4         NaN          1                NaN            NaN            NaN   

   NU_NOTA_COMP3  NU_NOTA_COMP4  NU_NOTA_COMP5  NOTA_REDACAO Q025  
0          180.0          200.0          200.0         940.0    B  
1            NaN            NaN            NaN           NaN    A  
2          200.0          200.0          200.0         880.0    A  
3          120.0          100.0          100.0         560.0    B  
4            NaN            NaN            NaN           NaN    B

In [ ]:
dados1 = dados1.rename(columns={'NU_NOTA_CN': 'NOTA_CN','NU_NOTA_CH': 'NOTA_CH',
                                'NU_NOTA_LC': 'NOTA_LC','NU_NOTA_MT': 'NOTA_MT'})

In [ ]:
dados1 = dados1.rename(columns={'NU_NOTA_COMP1': 'COMP1','NU_NOTA_COMP2': 'COMP2',
                                'NU_NOTA_COMP3': 'COMP3','NU_NOTA_COMP4': 'COMP4',
                                'NU_NOTA_COMP5':'COMP5'})

In [ ]:
dados1 = dados1.rename(columns={'NU_IDADE': 'IDADE','TP_SEXO': 'SEXO',
                                'TP_COR_RACA': 'RACA','Q025': 'INTERNET',
                                'TP_ESCOLA':'ESCOLA'})

Renomeando Registros (Linhas)

In [ ]:
dados1["RACA"] = dados1["RACA"].replace({0:"nao_declarado", 1:"branca", 2:"preta", 3:"parda", 4:"amarela", 5:"indigena"})

In [ ]:
dados1.head()

NU_INSCRICAO NO_MUNICIPIO_RESIDENCIA  IDADE SEXO  TP_ESTADO_CIVIL    RACA  \
0     1,90E+11            Santa Isabel     22    M                1   parda   
1     1,90E+11             Pinhalzinho     19    M                1   parda   
2     1,90E+11               São Paulo     17    F                1   parda   
3     1,90E+11   São Bernardo do Campo     19    M                1   parda   
4     1,90E+11    São Joaquim da Barra     37    M                2  branca   

   TP_NACIONALIDADE  TP_ST_CONCLUSAO  ESCOLA  IN_TREINEIRO   CO_ESCOLA  \
0                 1                1       1             0         NaN   
1                 1                1       1             0         NaN   
2                 1                2       3             0  15567761.0   
3                 1                2       2             0  35904958.0   
4                 1                1       1             0         NaN   

        NO_MUNICIPIO_ESC  TP_PRESENCA_CN  TP_PRESENCA_CH  TP_PRESENCA_LC  \
0                    NaN               1               1               1   
1                    NaN               0               0               0   
2                  Belém               1               1               1   
3  São Bernardo do Campo               1               1               1   
4                    NaN               0               0               0   

   TP_PRESENCA_MT  NOTA_CN  NOTA_CH  NOTA_LC  NOTA_MT  TP_LINGUA  \
0               1    564.6    585.8    592.9    600.2          0   
1               0      NaN      NaN      NaN      NaN          1   
2               1    576.2    641.1    634.3    731.8          0   
3               1    402.2    424.8    509.6    393.2          1   
4               0      NaN      NaN      NaN      NaN          1   

   TP_STATUS_REDACAO  COMP1  COMP2  COMP3  COMP4  COMP5  NOTA_REDACAO INTERNET  
0                1.0  160.0  200.0  180.0  200.0  200.0         940.0        B  
1                NaN    NaN    NaN    NaN    NaN    NaN           NaN        A  
2                1.0  160.0  120.0  200.0  200.0  200.0         880.0        A  
3                1.0   80.0  160.0  120.0  100.0  100.0         560.0        B  
4                NaN    NaN    NaN    NaN    NaN    NaN           NaN        B

In [ ]:
# OUTRA MANEIRA
# dados1['RACA'] = dados1['RACA'].replace([0,1,2,3,4,5], ['Não Declarado','Branca','Preta','Parda','Amarela','Indígena'])

In [ ]:
dados1["TP_LINGUA"] = dados1["TP_LINGUA"].replace({0:"Inglês", 1:"Espanhol"})

In [ ]:
dados1["ESCOLA"] = dados1["ESCOLA"].replace({1:"não_respondeu", 2:"pública", 3:"privada", 4:"exterior"})

In [ ]:
dados1["INTERNET"] = dados1["INTERNET"].replace({"A":"Não", "B":"Sim"})

## Análise Exploratória e Transformação dos dados

**Análise das idades**

In [ ]:
dados1['IDADE'].value_counts()


17    219313
18    147548
19     81236
16     59316
20     53009
       ...  
86         1
4          1
3          1
91         1
2          1
Name: IDADE, Length: 80, dtype: int64

In [ ]:
dados1['IDADE'].value_counts().sort_index()

In [ ]:
# Para ver todas as linhas, primeiro transforme num Dataframe:
idade = pd.DataFrame(dados1['IDADE'].value_counts().sort_index())

In [ ]:
# Configura o Pandas para observar 80 linhas
pd.set_option('display.max_rows', 80)

In [ ]:
idade.head(80)

Alguns valores estranhos e curiosos: idades menores que 12 anos e maiores de 80 anos.

In [ ]:
menores_12 = dados1.query('IDADE < 12')['NO_MUNICIPIO_RESIDENCIA'].value_counts()

In [ ]:
# Inscrições com idade abaixo de 12 anos serão excluídos devido a alta probabilidade de erro.
menores_12

In [ ]:
dados_maiores_11 = dados1.loc[dados1.IDADE > 11]

In [ ]:
dados_maiores_11.head()

In [ ]:
dados_maiores_11['IDADE'].value_counts().sort_index()

**Treineiros**

In [ ]:
dados_maiores_11.query('IN_TREINEIRO == 1')['IN_TREINEIRO'].value_counts()

1    93988
Name: IN_TREINEIRO, dtype: int64

In [ ]:
treineiros = dados_maiores_11.loc[dados_maiores_11.IN_TREINEIRO == 1]

In [ ]:
treineiros.head()

In [ ]:
treineiros.to_csv('treineiros_enem_2019_sp.csv', encoding = 'iso-8859-1', index = False)

In [ ]:
# Separação entre os treineiros e os vestibulandos
vestibulandos = dados_maiores_11.loc[dados_maiores_11.IN_TREINEIRO == 0]

In [ ]:
vestibulandos.head()

NU_INSCRICAO NO_MUNICIPIO_RESIDENCIA  IDADE SEXO  TP_ESTADO_CIVIL    RACA  \
0     1,90E+11            Santa Isabel     22    M                1   parda   
1     1,90E+11             Pinhalzinho     19    M                1   parda   
2     1,90E+11               São Paulo     17    F                1   parda   
3     1,90E+11   São Bernardo do Campo     19    M                1   parda   
4     1,90E+11    São Joaquim da Barra     37    M                2  branca   

   TP_NACIONALIDADE  TP_ST_CONCLUSAO         ESCOLA  IN_TREINEIRO   CO_ESCOLA  \
0                 1                1  não_respondeu             0         NaN   
1                 1                1  não_respondeu             0         NaN   
2                 1                2        privada             0  15567761.0   
3                 1                2        pública             0  35904958.0   
4                 1                1  não_respondeu             0         NaN   

        NO_MUNICIPIO_ESC  TP_PRESENCA_CN  TP_PRESENCA_CH  TP_PRESENCA_LC  \
0                    NaN               1               1               1   
1                    NaN               0               0               0   
2                  Belém               1               1               1   
3  São Bernardo do Campo               1               1               1   
4                    NaN               0               0               0   

   TP_PRESENCA_MT  NOTA_CN  NOTA_CH  NOTA_LC  NOTA_MT TP_LINGUA  \
0               1    564.6    585.8    592.9    600.2    Inglês   
1               0      NaN      NaN      NaN      NaN  Espanhol   
2               1    576.2    641.1    634.3    731.8    Inglês   
3               1    402.2    424.8    509.6    393.2  Espanhol   
4               0      NaN      NaN      NaN      NaN  Espanhol   

   TP_STATUS_REDACAO  COMP1  COMP2  COMP3  COMP4  COMP5  NOTA_REDACAO INTERNET  
0                1.0  160.0  200.0  180.0  200.0  200.0         940.0      Sim  
1                NaN    NaN    NaN    NaN    NaN    NaN           NaN      Não  
2                1.0  160.0  120.0  200.0  200.0  200.0         880.0      Não  
3                1.0   80.0  160.0  120.0  100.0  100.0         560.0      Sim  
4                NaN    NaN    NaN    NaN    NaN    NaN           NaN      Sim

In [ ]:
vestibulandos.shape

(719768, 29)

**Análise das presenças**

In [ ]:
vestibulandos['TP_PRESENCA_CN'].value_counts()

1    500805
0    218693
2       270
Name: TP_PRESENCA_CN, dtype: int64

In [ ]:
vestibulandos['TP_PRESENCA_CH'].value_counts()

1    533142
0    185969
2       657
Name: TP_PRESENCA_CH, dtype: int64

In [ ]:
vestibulandos['TP_PRESENCA_LC'].value_counts()

1    533142
0    185969
2       657
Name: TP_PRESENCA_LC, dtype: int64

In [ ]:
vestibulandos['TP._PRESENCA_MT'].value_counts()

1    500805
0    218693
2       270
Name: TP_PRESENCA_MT, dtype: int64

RESUMO DA PRESENÇA NAS PROVAS:

Ciências da Natureza: 500805 presentes, 218693 faltaram e 270 eliminados.

Ciências Humanas: 533142 presentes, 185969 faltaram e 657 eliminados.

Linguagens e Códigos: 533142 presentes, 185969 faltaram e 657 eliminados.

Matemática: 500805 presentes, 218693 faltaram e 270 eliminados.

In [ ]:
vestibulandos['TP_STATUS_REDACAO'].value_counts().sort_index()

1.0    519763
2.0       913
3.0      1417
4.0      5513
6.0      3626
7.0       543
8.0       828
9.0       539
Name: TP_STATUS_REDACAO, dtype: int64

**Valores Ausentes**

In [ ]:
vestibulandos['IDADE'].isnull().sum()

0

In [ ]:
provas = ['NOTA_REDACAO','NOTA_CN','NOTA_MT','NOTA_LC','NOTA_CH']

In [ ]:
# Registros NAN nas notas
vestibulandos[provas].isnull().sum()

NOTA_REDACAO    186626
NOTA_CN         218963
NOTA_MT         218963
NOTA_LC         186626
NOTA_CH         186626
dtype: int64

In [ ]:
vestibulandos_classificados = vestibulandos.loc[vestibulandos.TP_PRESENCA_CH == 1]
vestibulandos_classificados = vestibulandos_classificados.loc[vestibulandos_classificados.TP_PRESENCA_MT == 1]

In [ ]:
vestibulandos_classificados['TP_PRESENCA_MT'].value_counts()

1    499940
Name: TP_PRESENCA_MT, dtype: int64

In [ ]:
vestibulandos_classificados['TP_PRESENCA_CH'].value_counts()

1    499940
Name: TP_PRESENCA_CH, dtype: int64

In [ ]:
vestibulandos_classificados['TP_PRESENCA_CN'].value_counts()

1    499940
Name: TP_PRESENCA_CN, dtype: int64

In [ ]:
vestibulandos_classificados['TP_PRESENCA_LC'].value_counts()

1    499940
Name: TP_PRESENCA_LC, dtype: int64

In [ ]:
vestibulandos_classificados[provas].isnull().sum()

NOTA_REDACAO    0
NOTA_CN         0
NOTA_MT         0
NOTA_LC         0
NOTA_CH         0
dtype: int64

** Notas zeros na Redação**

In [ ]:
from collections import Counter
dict(Counter(vestibulandos_classificados['NOTA_REDACAO']))

In [ ]:
vestibulandos_classificados.query('NOTA_REDACAO == 0')['NOTA_REDACAO'].value_counts()

In [ ]:
vestibulandos_classificados.query('COMP1 == 0')['COMP1'].value_counts()

In [ ]:
vestibulandos_classificados.query('COMP2 == 0')['COMP2'].value_counts()

In [ ]:
vestibulandos_classificados.query('COMP3 == 0')['COMP3'].value_counts()

In [ ]:
vestibulandos_classificados.query('COMP4 == 0')['COMP4'].value_counts()

In [ ]:
vestibulandos_classificados.query('COMP5 == 0')['COMP5'].value_counts()

NOTAS ZEROS EM REDAÇÃO:

REDAÇÃO (final): 9181 notas zeros devido a não cumprimento das regras ou tirou zero em todos os cinco componentes.

COMPONENTE 1(Demonstrar domínio da modalidade escrita formal da Língua Portuguesa.): 9188 notas zeros somente neste item.

COMPONENTE 2 (Compreender a proposta de redação e aplicar conceitos das várias áreas de conhecimento para desenvolver o tema, dentro dos limites estruturais do texto dissertativo-argumentativo em prosa): 9181.

COMPONENTE 3 (Selecionar, relacionar, organizar e interpretar informações, fatos, opiniões e argumentos em defesa de um ponto de vista): 9197 notas zeros somente neste item.

COMPONENTE 4 (Demonstrar conhecimento dos mecanismos linguísticos necessários para a construção da argumentação.): 9194 notas zeros somente neste item.

COMPONENTE 5 (Elaborar proposta de intervenção para o problema abordado, respeitando os direitos humanos): 73330 notas zeros somente neste item.


In [ ]:
vestibulandos_classificados.query('NOTA_CH == 0')['NOTA_CH'].value_counts()

In [ ]:
vestibulandos_classificados.query('NOTA_CN == 0')['NOTA_CN'].value_counts()

In [ ]:
vestibulandos_classificados.query('NOTA_LC == 0')['NOTA_LC'].value_counts()

In [ ]:
vestibulandos_classificados.query('NOTA_MT == 0')['NOTA_MT'].value_counts()

In [ ]:
vestibulandos_classificados.query('NOTA_REDACAO == 0')['NOTA_REDACAO'].value_counts()

NOTAS ZEROS DOS INSCRITOS CONCORRENTES AO VESTIBULAR:

Matemática = 56

Ciências da Natureza = 36

Linguagens e Códigos = 135

Ciências Humanas = 343

Redação = 9181


In [ ]:
vestibulandos_classificados.head()

In [ ]:
vestibulandos_classificados = vestibulandos_classificados.drop(columns=['IN_TREINEIRO'], )

In [ ]:
vestibulandos_classificados.shape

## **Salvando (Exportando) o Dataframe Tratado**

In [ ]:
vestibulandos_classificados.to_csv('enem_2019_tratado.csv', encoding = 'iso-8859-1', index = False)